base_multi_company/models/multi_company_abstract.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class MultiCompanyAbstract(models.AbstractModel):
+
+    _name = 'multi.company.abstract'
+    _description = 'Multi-Company Abstract'
+
+    company_id = fields.Many2one(
+        string='Company',
+        comodel_name='res.company',
+        compute='_compute_company_id',
+        inverse='_inverse_company_id',
+        search='_search_company_id',
+    )
+    company_ids = fields.Many2many(
+        string='Companies',
+        comodel_name='res.company.assignment',
+        default=lambda s: s._default_company_ids(),
```

Comment:
@lasley @pedrobaeza IMO we should add `auto_join=True` to generate JOINs upon search through that field. This will improve the performances and it's safe since record rules doesn't apply/matter on this relation.